3/test/excercise3.spec.js

```diff
@@ -0,0 +1,110 @@
+const { gql } = require('apollo-server')
+const server = require('../server.js')
+const {createTestClient} = require('apollo-server-testing');
+const jwt = require('jsonwebtoken');
+
+const { query, mutate } = createTestClient(server);
+
+describe('Query: Given a list of todos and assignees', () => {
+
+    const queryTodoLength = gql`query getCountTodos {countTodos}`;
+    const queryAssigneeLength = gql`query getCountAssignees {countAssignees}`;
+    
+    it('get all todo items at startup', () => {
+        query({query: queryTodoLength}).then((result) => {
+            expect(result.data.countTodos).toBe(6);
+        })
+    })
+    it('get all assignees at startup', () => {
+        query({query: queryAssigneeLength}).then((result) => {
+            expect(result.data.countAssignees).toBe(3);
+        })
+    })
+
+    const querySignedAssignee = gql`query getSigneAssignee($token: String!) { assignees(token: $token) {id, name} }`;   
+    const tokenMessage = jwt.sign({auth: 'der richtige inhalt'}, 'shhhhh');
+
+    it('get all assignees signed', () => {
+        query({query: querySignedAssignee, variables: {token: tokenMessage}}).then((result) => {
```

Comment:
As I said above, I would not pass in the token via `args`. Instead you can pass it in via request headers.

In a test case, you would expect a successfully decoded token by overriding the `context` hook in `createTestClient`:

```js
let authenticatedUser = null // override this user in your test cases before you call `query` or `mutate`

const server = createServer({ context: () => ({ authenticatedUser })})
// this is a custom helper function wich returns an apollo server with an overridden context hook

const { query, mutate } = createTestClient(server);
```